app/src/main/java/com/celzero/bravedns/ui/CustomIpActivity.kt

```diff
@@ -15,44 +15,46 @@
  */
 package com.celzero.bravedns.ui
 
-import android.app.Activity
 import android.app.Dialog
+import android.content.Context
+import android.content.res.Configuration
 import android.os.Bundle
 import android.view.View
 import android.view.Window
 import android.view.WindowManager
 import android.widget.Toast
 import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.SearchView
 import androidx.core.view.isVisible
 import androidx.core.widget.addTextChangedListener
-import androidx.lifecycle.LifecycleOwner
-import androidx.recyclerview.widget.LinearLayoutManager
 import androidx.recyclerview.widget.RecyclerView
+import by.kirich1409.viewbindingdelegate.viewBinding
 import com.celzero.bravedns.R
 import com.celzero.bravedns.adapter.CustomIpAdapter
 import com.celzero.bravedns.automaton.IpRulesManager
+import com.celzero.bravedns.databinding.ActivityCustomIpBinding
 import com.celzero.bravedns.databinding.DialogAddCustomIpBinding
-import com.celzero.bravedns.databinding.DialogCustomIpBinding
+import com.celzero.bravedns.service.PersistentState
+import com.celzero.bravedns.util.CustomLinearLayoutManager
+import com.celzero.bravedns.util.Themes
 import com.celzero.bravedns.util.Utilities
 import com.celzero.bravedns.viewmodel.CustomIpViewModel
 import inet.ipaddr.IPAddressString
+import org.koin.android.ext.android.inject
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class CustomIpActivity : AppCompatActivity(R.layout.activity_custom_ip),
+                         SearchView.OnQueryTextListener {
 
-class CustomIpDialog(val activity: Activity, val viewModel: CustomIpViewModel, themeId: Int) :
-        Dialog(activity, themeId), SearchView.OnQueryTextListener {
     private var layoutManager: RecyclerView.LayoutManager? = null
-    private lateinit var b: DialogCustomIpBinding
-    private lateinit var adapter: CustomIpAdapter
+    private val b by viewBinding(ActivityCustomIpBinding::bind)
+    private val viewModel: CustomIpViewModel by viewModel()
+    private val persistentState by inject<PersistentState>()
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        setTheme(Themes.getCurrentTheme(isDarkThemeOn(), persistentState.theme))
         super.onCreate(savedInstanceState)
-        setCancelable(true)
-        requestWindowFeature(Window.FEATURE_NO_TITLE)
-        b = DialogCustomIpBinding.inflate(layoutInflater)
-        setContentView(b.root)
-
-        window?.setLayout(WindowManager.LayoutParams.MATCH_PARENT,
```

Comment:
curious: why remove?

Reply:
Its moved from Dialog to Activity, Activity do not require window properties.